File: Chris/2022-06-07-BestTimeToBuyAndSellStock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        lowest = prices[0]
        maxProfit = 0
        
        for price in prices:
            if(price-lowest > maxProfit ):
                maxProfit = price-lowest
            
            if(lowest > price):
                lowest = price
        
        
        return maxProfit

File: Chris/2022-06-07-BestTimeToBuyAndSellStock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int buyPrice = prices[0];
        int maxProfit = 0;
        
        
        foreach(int p in prices){
            
            if(p - buyPrice > maxProfit){
                maxProfit = p - buyPrice;
            }
            
            if(p < buyPrice){
                buyPrice = p;
            }
        }
        
        return maxProfit;
    }
}